src/libraries/System.Reflection.Metadata/tests/Metadata/TypeNameTests.cs

```diff
@@ -493,6 +581,26 @@ public void GetNodeCountReturnsExpectedValue(Type type, int expected)
             }
         }
 
+        [OuterLoop("It takes a lot of time")]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.Is64BitProcess))] // 32 is likely to OOM
+        public void GetNodeCountThrowsForInt32Overflow()
+        {
+            TypeName genericType = TypeName.Parse("Generic".AsSpan());
+            TypeName genericArg = TypeName.Parse("Arg".AsSpan());
+            // Don't allocate Array.MaxLength array as it may make this test flaky (frequent OOMs).
+            ImmutableArray<TypeName>.Builder genericArgs = ImmutableArray.CreateBuilder<TypeName>(initialCapacity: int.MaxValue / 64);
```

Comment:
This does not need to allocate very large array. Instead, it can repeat MakeGenericTypeName multiple times. It will make the test a lot less resource intensive, faster and compatible with 32-bit platforms.

Reply:
> This does not need to allocate very large array. Instead, it can repeat MakeGenericTypeName multiple times.

I know and in my opinion this is exactly what it's doing right now. But I assume that `int.MaxValue / 64` is still very large array for you. What would not be? `int.MaxValue / 1024`?

Reply:
It can be very small, like 10.

Reply:
`Math.Sqrt(int.MaxValue)` does what we need, I've pushed the change